db/db_options_test.cc

```diff
@@ -1033,6 +1034,137 @@ TEST_F(DBOptionsTest, SetFIFOCompactionOptions) {
   ASSERT_EQ(fifo_temp_opt[1].age, 30000);
 }
 
+TEST_F(DBOptionsTest, OffPeakTimes) {
+  Options options;
+  options.create_if_missing = true;
+
+  auto verify_invalid = [&]() {
+    Status s = DBImpl::TEST_ValidateOptions(options);
+    ASSERT_NOK(s);
+    ASSERT_TRUE(s.IsInvalidArgument());
+  };
+
+  auto verify_valid = [&]() {
+    Status s = DBImpl::TEST_ValidateOptions(options);
+    ASSERT_OK(s);
+    ASSERT_FALSE(s.IsInvalidArgument());
+  };
+  std::vector<std::string> invalid_cases = {
+      "06:30-",         "-23:30",           // Both need to be set
+      "12:30 PM-23:30", "12:01AM-11:00PM",  // Invalid format
+      "01:99-25:00",  // Invalid value for the hours or minutes
+      "random-value",   "No:No-Hi:Hi",
+  };
+
+  std::vector<std::string> valid_cases = {
+      "",  // Not enabled. Valid case
+      "06:30-11:30",
+      "06:30-23:30",
+      "13:30-14:30",
+      "23:30-01:15",  // From 11:30PM to 1:15AM next day. Valid case
```

Comment:
Like I mentioned in the earlier comment, we've decided to have multiple offpeak periods to avoid unnecessary complexity. I think our API is clear enough for user to understand the overnight timespan.

If there's a strong need of multiple offpeak periods later, we can add support for different format as you suggested (a string with multiple dates and a well-known separator)